app/presenters/describe_repair.rb

```diff
@@ -0,0 +1,38 @@
+class DescribeRepair
+  attr_reader :form
+
+  def initialize(form_params: {}, details:, answers:, partial_checker:)
+    @form_params = form_params
+    @answers = answers
+    @details = details
+    @partial_checker = partial_checker
+  end
+
+  def partial
```

Comment:
Just playing with this through the frontend, and I was wondering whether this should be...

```ruby
  def partial
    return @details if @partial_checker.exists?(@details)

    if diagnosed?
      'anything_else'
    else
      'describe_problem'
    end
  end
```

i.e. Always use the partial referenced in the `desc` key, but failover to `anything_else` or `describe_problem` depending on whether we have a SOR code or not(?)

Reply:
Agreed - good fix. The problem will still be that the form should be required, but will be optional, but that's a different problem - fixed by resetting the session.